> I don't see why the Callable couldn't be created. Try `callable_mp(RSG::particles_storage, &RendererParticlesStorage::particles_set_view_axis)`.

I tried that, results in this error:

`.\core/object/callable_method_pointer.h(116): error C2039: 'get_instance_id': is not a member of 'RendererParticlesStorage'
.\servers/rendering/storage/particles_storage.h(36): note: see declaration of 'RendererParticlesStorage'
.\core/object/callable_method_pointer.h(116): note: the template instantiation context (the oldest one first) is
servers\rendering\renderer_scene_cull.cpp(3022): note: see reference to function template instantiation 'Callable create_custom_callable_function_pointer<RendererParticlesStorage,RID,const Vector3&,const Vector3&>(T *,const char *,void (__cdecl RendererParticlesStorage::* )(RID,const Vector3 &,const Vector3 &))' being compiled
        with
        [
            T=RendererParticlesStorage
        ]
.\core/object/callable_method_pointer.h(129): note: see reference to class template instantiation 'CallableCustomMethodPointer<RendererParticlesStorage,void,RID,const Vector3 &,const Vector3 &>' being compiled
.\core/object/callable_method_pointer.h(113): note: while compiling class template member function 'CallableCustomMethodPointer<RendererParticlesStorage,void,RID,const Vector3 &,const Vector3 &>::CallableCustomMethodPointer(T *,R (__cdecl RendererParticlesStorage::* )(RID,const Vector3 &,const Vector3 &))'
        with
        [
            T=RendererParticlesStorage,
            R=void
        ]
.\core/object/callable_method_pointer.h(129): note: see the first reference to 'CallableCustomMethodPointer<RendererParticlesStorage,void,RID,const Vector3 &,const Vector3 &>::CallableCustomMethodPointer' in 'create_custom_callable_function_pointer'`

--- a/servers/rendering/renderer_scene_cull.cpp
+++ b/servers/rendering/renderer_scene_cull.cpp
@@ -3017,7 +3017,8 @@ void RendererSceneCull::_scene_cull(CullData &cull_data, InstanceCullResult &cul
 							cull_data.cull->lock.lock();
 							RSG::particles_storage->particles_request_process(idata.base_rid);
 							cull_data.cull->lock.unlock();
-							RSG::particles_storage->particles_set_view_axis(idata.base_rid, -cull_data.cam_transform.basis.get_column(2).normalized(), cull_data.cam_transform.basis.get_column(1).normalized());
+
+							RS::get_singleton()->call_on_render_thread(callable_mp_static(&RendererSceneCull::_scene_particles_set_view_axis).bind(idata.base_rid, -cull_data.cam_transform.basis.get_column(2).normalized(), cull_data.cam_transform.basis.get_column(1).normalized()));
 							//particles visible? request redraw
 							RenderingServerDefault::redraw_request();
 						}
@@ -3190,6 +3191,10 @@ void RendererSceneCull::_scene_cull(CullData &cull_data, InstanceCullResult &cul
 			cull_result.mesh_instances.push_back(cull_data.scenario->instance_data[i].instance->mesh_instance);
 		}
 	}
+}
+
+void RendererSceneCull::_scene_particles_set_view_axis(RID p_particles, const Vector3 &p_axis, const Vector3 &p_up_axis) {
+	RSG::particles_storage->particles_set_view_axis(p_particles, p_axis, p_up_axis);
 }
 
 void RendererSceneCull::_render_scene(const RendererSceneRender::CameraData *p_camera_data, const Ref<RenderSceneBuffers> &p_render_buffers, RID p_environment, RID p_force_camera_attributes, RID p_compositor, uint32_t p_visible_layers, RID p_scenario, RID p_viewport, RID p_shadow_atlas, RID p_reflection_probe, int p_reflection_probe_pass, float p_screen_mesh_lod_threshold, bool p_using_shadows, RenderingMethod::RenderInfo *r_render_info) {

--- a/servers/rendering/renderer_scene_cull.h
+++ b/servers/rendering/renderer_scene_cull.h
@@ -1184,9 +1184,11 @@ public:
 
 	void _scene_cull_threaded(uint32_t p_thread, CullData *cull_data);
 	void _scene_cull(CullData &cull_data, InstanceCullResult &cull_result, uint64_t p_from, uint64_t p_to);
+	static void _scene_particles_set_view_axis(RID p_particles, const Vector3 &p_axis, const Vector3 &p_up_axis);
 	_FORCE_INLINE_ bool _visibility_parent_check(const CullData &p_cull_data, const InstanceData &p_instance_data);
 
 	bool _render_reflection_probe_step(Instance *p_instance, int p_step);
+
 	void _render_scene(const RendererSceneRender::CameraData *p_camera_data, const Ref<RenderSceneBuffers> &p_render_buffers, RID p_environment, RID p_force_camera_attributes, RID p_compositor, uint32_t p_visible_layers, RID p_scenario, RID p_viewport, RID p_shadow_atlas, RID p_reflection_probe, int p_reflection_probe_pass, float p_screen_mesh_lod_threshold, bool p_using_shadows = true, RenderInfo *r_render_info = nullptr);
 	void render_empty_scene(const Ref<RenderSceneBuffers> &p_render_buffers, RID p_scenario, RID p_shadow_atlas);
 

